tests/FSharp.Compiler.Service.Tests/LibraryTestFx.fs

```diff
@@ -54,77 +54,19 @@ module SurfaceArea =
-            |> set
-
-        //
-        // Find types/members which exist in exactly one of the expected or actual surface areas.
-        //
-
-        /// Surface area types/members which were expected to be found but missing from the actual surface area.
-        let unexpectedlyMissing = Set.difference expected actual
-
-        /// Surface area types/members present in the actual surface area but weren't expected to be.
-        let unexpectedlyPresent = Set.difference actual expected
-
-        // If both sets are empty, the surface areas match so allow the test to pass.
-        if Set.isEmpty unexpectedlyMissing
-          && Set.isEmpty unexpectedlyPresent then
-            // pass
-            ()
-        else
-
-            let logFile =
-                let workDir = TestContext.CurrentContext.WorkDirectory
-                sprintf "%s\\FSharp.CompilerService.SurfaceArea.%s.txt" workDir platform
-
-            FileSystem.OpenFileForWriteShim(logFile).Write(String.Join("\r\n", actual))
-
-            // The surface areas don't match; prepare an easily-readable output message.
-            let msg =
-                let inline newLine (sb : System.Text.StringBuilder) = sb.AppendLine () |> ignore
-                let sb = System.Text.StringBuilder ()
-                Printf.bprintf sb "Assembly: %A" asm
-                newLine sb
-                sb.AppendLine "Expected and actual surface area don't match. To see the delta, run:" |> ignore
-                Printf.bprintf sb "    windiff %s %s" baseline logFile
-                newLine sb
-                newLine sb
-                sb.AppendLine "To update the baseline copy the contents of this:" |> ignore
-                Printf.bprintf sb "    %s" logFile
-                newLine sb
-                sb.AppendLine "into this:" |> ignore
-                Printf.bprintf sb "    %s" baseline
-                newLine sb
-                newLine sb
-                sb.Append "Unexpectedly missing (expected, not actual):" |> ignore
-                for s in unexpectedlyMissing do
-                    newLine sb
-                    sb.Append "    " |> ignore
-                    sb.Append s |> ignore
-                newLine sb
-                newLine sb
-                sb.Append "Unexpectedly present (actual, not expected):" |> ignore
-                for s in unexpectedlyPresent do
-                    newLine sb
-                    sb.Append "    " |> ignore
-                    sb.Append s |> ignore
-                newLine sb
-                sb.ToString ()
-
-            failwith msg
+    let verify expectedFile actualFile =
+      let normalize text = Regex.Replace(text, "(\\r\\n|\\n|\\r)+", "\r\n").Trim()
+      let _asm, actual = getActual ()
+      let actual = normalize actual
+      let expected = normalize (System.IO.File.ReadAllText expectedFile)
+      if expected <> actual then
+          System.IO.File.WriteAllText(actualFile, actual)
```

Comment:
All filesystem operations should be done via the FileSystem, could you please change it? There should be some extensions to stream to read/write all.

Reply:
@vzarytovskii understood, sorry as I wanted to move the file to diff separately from the test.

I'll look at fixing what you point, using the component you describe.

Otherwise, is there any issue with what I suggest to do? 

It makes it easier to diff/merge, run test again, and track the changes to have a separate file.